### PR TITLE
feat(api,web): adopt negative path testing standard

### LINKED ISSUE #384: Adopt negative path testing standard from ops/standards/testing.md
## Summary
The testing standard now includes a Negative Path Testing section (added 2026-04-08). This issue tracks adoption in the mokumo codebase.

## Context
Identified as structural root cause of the CodeRabbit routing bug (2026-04-08). The standard was authored and merged. Now the mokumo repo needs to implement the requirements.

## Acceptance Criteria
- [x] Standard authored in ops/standards/testing.md (completed 2026-04-08)
- [ ] Routing contract test suite created at `tests/api/routing/`
- [ ] Pre-implementation boundary checklist referenced in CLAUDE.md or contributing guide
- [ ] Existing endpoints audited for negative path coverage gaps
- [ ] At least one example of frontend negative path tests (error boundary)

Blocked by #376 (branded error page — needed for error boundary testing)

### PR DESCRIPTION
**Pull request by cmbays**
## Summary

Adopts the negative path testing standard from `ops/standards/testing/negative-path.md` into the Mokumo codebase.

- **Routing contract test suite** at `tests/api/routing/` — Hurl tests proving unknown `/api/*` paths return JSON 404 (not SPA HTML) and wrong HTTP methods return JSON 405
- **`method_not_allowed_fallback`** on the Axum router returning structured JSON 405 with the standard `ErrorBody` shape
- **Pre-implementation boundary checklist** added as coding standard #16 in CLAUDE.md
- **Frontend error boundary tests** — Playwright BDD scenarios verifying `+error.svelte` renders branded 404 page
- **Expanded `serve_spa` unit tests** — trailing slash, prefix collision, Content-Type assertions
- **`MethodNotAllowed` ErrorCode variant** added to `crates/types/src/error.rs`

### Endpoint Negative Path Coverage Audit

| Category | Before | After | Notes |
|---|---|---|---|
| Routing contract (404 JSON) | 0 Hurl tests | 5 assertions | `api-prefix-boundary.hurl` |
| Method rejection (405 JSON) | 0 Hurl tests | 5 assertions | `method-rejection.hurl` |
| Auth failures (401) | 3 Hurl tests | 3 Hurl tests | Existing coverage adequate |
| Entity not found (404) | 1 Hurl test | 1 Hurl test | `not-found.hurl` covers zero-UUID |
| Malformed body (422) | 0 | 0 | Follow-up: #TBD |
| Boundary IDs | 0 | 0 | Follow-up: #TBD |
| Frontend error boundary | 0 | 2 scenarios | `error-boundary.feature` |

Closes #384

## Test plan

- [ ] `moon run api:test` passes (new unit tests for `serve_spa` boundary cases and `handle_method_not_allowed`)
- [ ] `moon run api:smoke` passes (new routing Hurl tests run against live server)
- [ ] `moon run api:lint` passes (new code passes clippy)
- [ ] `moon run web:test` passes (new error boundary BDD scenarios)
- [ ] Verify `POST /api/health` returns 405 JSON (not HTML)
- [ ] Verify `GET /api/nonexistent` returns 404 JSON (not SPA HTML)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Unknown /api/* routes now return structured JSON 404; unsupported HTTP methods on API endpoints return structured JSON 405. API-prefix collisions still serve non-API responses.

* **Tests**
  * Added routing and method-rejection contract tests plus end-to-end checks for the branded error page and updated BDD test inclusion/exclusion.

* **Documentation**
  * Updated changelog and added a pre-implementation boundary checklist rule.

* **Types**
  * Added a new machine-readable error code: "method_not_allowed".

### CODE REVIEW

File: services/api/src/lib.rs [hunk 1]
```diff
@@ -1061,6 +1062,20 @@ fn spa_response(status: StatusCode, content_type: &str, cache: &str, body: Vec<u
...
+        message: "Method not allowed".into(),
+        details: None,
+    };
+    (
+        StatusCode::METHOD_NOT_ALLOWED,
+        [(axum::http::header::CACHE_CONTROL, "no-store")],
+        Json(body),
+    )
+        .into_response()
+}
```
[the remark this answers is not in the document]
Comment (cmbays) [reply services/api/src/lib.rs:1077]: Good catch on RFC 9110 §15.5.6. This is a known limitation of Axum's `method_not_allowed_fallback` API — the handler receives no information about which methods are registered for the matched route, so it cannot populate the `Allow` header.

The trade-off here:
- **Default Axum 405** includes the `Allow` header but returns an empty body (not JSON), breaking our `{"code", "message", "details"}` contract.
- **Custom `method_not_allowed_fallback`** returns our standard JSON error shape but loses `Allow`.

We chose JSON error shape consistency since this is a self-hosted product with no third-party API consumers relying on `Allow` for content negotiation. The middleware-interception approach (intercept default 405, rewrite body to JSON, preserve `Allow`) would work but adds complexity disproportionate to the benefit here.

Filed as a known gap — if we ever publish a public API, we should revisit with the middleware approach.

Re: the `ErrorBody` duplication between `handle_method_not_allowed` and `serve_spa` — agreed that these could share a helper, but they serve different error codes and messages with no shared parameterization beyond the struct. A helper would just be `ErrorBody { code, message, details: None }` which is the struct literal itself. Deferring until a third call site emerges.
